src/Components/MapComponent/MapComponent.jsx

```diff
@@ -147,9 +162,116 @@ class MapComponent extends Component {
       onSetCurrentStopsGeoJSON(newCurentStopsGeoJSON);
     });
 
+    const modify = new Modify({
+      source: this.routeVectorSource,
+      pixelTolerance: 2,
+      style: () => {
+        const { currentMot } = this.props;
+        return pointStyleFunction(currentMot);
+      },
+    });
+
+    modify.on('modifystart', evt => {
+      // save start point to find where to add the new HOP!
+      this.initialRouteDrag = {
+        features: evt.features.getArray(),
+        coordinate: evt.mapBrowserEvent.coordinate,
+      };
+    });
+
+    modify.on('modifyend', evt => {
+      const { features } = this.initialRouteDrag;
+      const {
+        currentMot,
+        currentStops,
+        currentStopsGeoJSON,
+        onSetCurrentStops,
+        onSetCurrentStopsGeoJSON,
+      } = this.props;
+      const updatedCurrentStops = _.clone(currentStops);
+      const updatedCurrentStopsGeoJSON = _.clone(currentStopsGeoJSON);
+      let newHopIdx = -1;
+
+      // No drag for foot/car for now on.
+      if (!GRAPHHOPPER_MOTS.includes(currentMot)) {
+        const flatCoords = features
+          .map(f => f.getGeometry().getFlatCoordinates())
+          .map(coords => {
+            return [
+              coords[0],
+              coords[1],
+              coords[coords.length - 2],
+              coords[coords.length - 1],
+            ];
+          });
+
+        const closestSegment = this.routeVectorSource
+          .getClosestFeatureToCoordinate(this.initialRouteDrag.coordinate)
+          .getGeometry()
+          .getFlatCoordinates();
+        const closestEdges = [
+          closestSegment[0],
+          closestSegment[1],
+          closestSegment[closestSegment.length - 2],
+          closestSegment[closestSegment.length - 1],
+        ];
```

Comment:
here too,  use getFirstCoordinate and getLastCoordinate

Reply:
Code updated, thanks it is more readable :)